gilbreth_perception/src/segmentation_node.cpp

```diff
@@ -0,0 +1,124 @@
+#include <ros/ros.h>
+#include <sensor_msgs/PointCloud2.h>
+#include <pcl_conversions/pcl_conversions.h>
+#include <pcl/point_types.h>
+#include <pcl/point_cloud.h>
+#include <pcl/segmentation/extract_clusters.h>
+#include <pcl/filters/voxel_grid.h>
+#include <pcl/filters/extract_indices.h>
+#include <pcl/filters/passthrough.h>
+#include <pcl/kdtree/kdtree_flann.h>
+#include <pcl/console/parse.h>
+
+
+
+
+
+//Algorithm params
+float down_sample_(0.01);
+
+ros::Publisher pub;
+
+void
+parseCommandLine (int argc, char *argv[])
+{
+	//General parameters
+	pcl::console::parse_argument (argc, argv, "--down_sample", down_sample_);
+
+}
+
+void
+cloud_cb (const sensor_msgs::PointCloud2ConstPtr& cloud_msg,int argc, char** argv)
```

Comment:
Why not parse the arguments at the beginning of the program instead of every time the callback is invoked?